### PR TITLE
[rapids] Updating RAPIDS to 0.18 and removing force uninstall of 'icu'

### PR DESCRIPTION
**Pull request by bradmiro**
Updating RAPIDs to 0.18. This update resolves a dependency conflict that previously required force uninstalling the `icu` dependency. 

Should also resolve https://github.com/GoogleCloudDataproc/initialization-actions/pull/860

### CODE REVIEW

**Comment by bradmiro** [comment]
Updated the version in the wrong spot, should work now...

Spark jars have not been released yet. @mengdong @sameerz  if you have any info on when 0.18 jars will be released that would be great.

**Comment by sameerz** [comment]
We are testing the 0.18 jar now and expect to have it available by Monday.  Can we hold off on updating this until we can also update `readonly DEFAULT_SPARK_RAPIDS_VERSION="0.3.0"` to 0.4.0, which will be sometime around March 10?  

The Spark RAPIDS 0.3.0 version is not tested with 0.18, but 0.4.0 will be.

**Comment by bradmiro** [comment]
Given that both Spark and Dask are currently broken anyway, I'm inclined to say we should just merge this once the Spark jars are available as this PR will at least fix Dask. We can update the Spark RAPIDS version as soon as it becomes available.

**Comment by bradmiro** [comment]
@sameerz @mengdong wanted to see if we're able to proceed here?

**Comment by mengdong** [comment]
Hi @bradmiro , we will update to 0.18 when v0.4 release of spark rapids is out. In meanwhile, dask rapids could work with cudf 0.18.

**Comment by bradmiro** [comment]
@mengdong ok, I modified this such that RAPIDS `0.18` is used with Dask and RAPIDS `0.17` is used with Spark so we can merge this and fix Dask. When the jars are ready, we can bump Spark to 0.18.

**Comment by bradmiro** [comment]
@medb the Spark tests will remain broken until Spark-RAPIDS v0.4 is released. Should I adjust the tests to use 2.0.0-RC22 (latest image with Spark 3.0.x) instead of latest, for the time being? Or should we just merge?

**Comment by medb** [comment]
Let's just merge then.

**Review by medb — APPROVED** [review]

**Pull request merged**
